crates/controller/src/hud_interaction.rs

```diff
@@ -0,0 +1,32 @@
+use bevy::prelude::*;
+
+use de_core::objects::BuildingType;
+use de_spawner::ObjectCounter;
+use crate::{
+    command::place_draft_system,
+    draft::NewDraftEvent,
+    pointer::Pointer,
+};
+
+#[derive(Component, Clone, Copy)]
+pub(crate) enum HudButtonAction {
+    Build(BuildingType),
+}
+
+pub(crate) fn hud_button_system(
+    interactions: Query<(&Interaction, &HudButtonAction), Changed<Interaction>>,
+    counter: Res<ObjectCounter>,
```

Comment:
Ditto, with this modification, it would make sense to move the counter logic into the receiver of the draft event.